flink-runtime/src/test/java/org/apache/flink/runtime/state/ttl/mock/MockKeyedStateBackend.java

```diff
@@ -178,7 +179,7 @@ public void notifyCheckpointComplete(long checkpointId) {
 	public RunnableFuture<SnapshotResult<KeyedStateHandle>> snapshot(
 		long checkpointId,
 		long timestamp,
-		@Nonnull CheckpointStreamFactory streamFactory,
+		@Nullable CheckpointStreamFactory streamFactory,
```

Comment:
I don't think this is the right direction to fix this.
Could you please share why `streamFactory` is null here?

Reply:
Thanks for review !
 In [MockStateBackend](https://github.com/apache/flink/blob/master/flink-runtime/src/test/java/org/apache/flink/runtime/state/ttl/mock/MockStateBackend.java) when call `snapshot` function it always create a `CheckpointStorage` object with  a `resolveCheckpointStorageLocation` function.In `resolveCheckpointStorageLocation` it only return null back. Maybe just fix this `resolveCheckpointStorageLocation` function in  [MockStateBackend](https://github.com/apache/flink/blob/master/flink-runtime/src/test/java/org/apache/flink/runtime/state/ttl/mock/MockStateBackend.java)  will better. What do you think.

Reply:
I can't reproduce your problem locally, All the tests passed on my laptop.
But I think If this failed on your machine, we may better make the `streamFactory` not null instead of marking it as `@Nullable`. So yes, as you analyzed, we may fix `MockStateBackend#resolveCheckpointStorageLocation`.

Reply:
Thank you for reply. I wonder if you turn on run time assertion in intellij as I mentioned in [Jira](https://issues.apache.org/jira/browse/FLINK-13188). In my computer, if I close this assertion detected button and rebuild project, I also can pass all the tests.

By the ways, I put a new commit to fix `MockStateBackend#resolveCheckpointStorageLocation`,  could you help have a look .